### PR TITLE
Feat/702 implement history tree storage

### PR DESCRIPTION
**Pull request by ognjenkurtic**
# Description

extends the workstep execution to store the new state tree root as a leaf in the history  tree

## Related Issue

#702 

## Motivation and Context

You need to know the history in order not to repeat it

## How Has This Been Tested

Will be covered with the e2e test and afterwards,  in mil5, with  unit tests
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Comment by ognjenkurtic** [comment]
Please review this after https://github.com/eea-oasis/baseline/pull/732 has been merged and this one rebased on top. I ll send a notification here once it is ready

File: examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ export class BpiAccountStorageAgent extends PrismaService {
             ownerBpiSubject: true,
           },
         },
+        // look into merkle tree profile to resolve unmarhsling of the state tree here
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts]: @ognjenkurtic do you want to list that as an open issue or close out before the PR is merged? At least, list as a TODO comment, please.

**Comment by ognjenkurtic** [comment]
@Kasshern @biscuitdey @Therecanbeonlyone1969 rebased, ready for review

**Review by Therecanbeonlyone1969 — COMMENTED** [review]
@ognjenkurtic looks good. left one minor item. then it is good to merge!

**Comment by ognjenkurtic** [comment]
> @ognjenkurtic looks good. left one minor item. then it is good to merge!

I removed the comment since it is obsolete. Resolved as part of next PR.

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Comment by ognjenkurtic** [comment]
@biscuitdey @Kasshern @Ybittan please review

**Review by biscuitdey — APPROVED** [review]
LGTM

**Pull request merged**